pyanaconda/modules/payloads/payload/rpm_ostree/installation.py

```diff
@@ -633,6 +681,155 @@ def _set_kargs(self):
+
+        # Some workarounds are needed due to current bootc development status
+
+        # The main one is SELinux to be presented in the system.
+        # https://github.com/bootc-dev/bootc/issues/1438
+        # It may be disabled but need to be presented.
+        # Both pairs of kernel parameters will work:
+        # * selinux=0 enforcing=0
+        # * selinux=1 enforcing=0
+        # Right now we use the /etc/selinux/config file to do the equivalent of
+        # selinux=1 enforcing=0.
+        # We set this in lorax-build using the whitelist_selinux.patch
+
+        # Bootc expects `prepare-root.conf` file to be presented in the system
+        # https://github.com/bootc-dev/bootc/discussions/1400
+        # https://github.com/bootc-dev/bootc/issues/1410
+        # echo -e "[ostree] \nsysroot=/sysroot" > /etc/ostree/prepare-root.conf
+        log.debug("Bootc workaround: add missing configuration file")
+        if not os.path.exists("/etc/ostree/prepare-root.conf"):
+            with open("/etc/ostree/prepare-root.conf", "w") as f:
+                f.write("[ostree]\n")
+                f.write("sysroot=/sysroot\n")
+        else:
+            log.debug("/etc/ostree/prepare-root.conf already presented and will not be modified")
+
+        # After automatic partitioning sysroot and sysimage are mounted,
+        # but we need a clear directory strucutre expected by the bootc
+        log.debug("Bootc workaround: remove unwanted mounts")
+        # umount -l /mnt/sysimage/
+        safe_exec_program("umount", ["-l", self._physroot])
+
+        # Bootc does not need any directories created automatically
+        # during the partitioning by blivet
+        log.debug("Bootc workaround: remove unwanted directories")
+        # rm -rf /mnt/sysroot/*
+        shutil.rmtree(self._sysroot + "/root")
+        directories_to_remove = ("dev", "proc", "run", "sys", "tmp", "home")
+        for directory in (f"{self._sysroot}/{d}" for d in directories_to_remove):
+            try:
+                os.rmdir(directory)
+            except FileNotFoundError:
+                # This is fine because we just need to make sure all of these
+                # directories do not exist
+                log.debug("No directory to remove: %s", directory)
+
+        # Bootc requires empty `boot` directory to be presentd
+        log.debug("Bootc workaround: create bootc required dirs")
+        # mkdir /mnt/sysroot/boot
+        # Security risk of exist_ok is allowed because the installer is
+        # running in a single user environment.
+        os.makedirs(self._sysroot + "/boot", mode=0o555, exist_ok=True)
+        # Mount /boot partition created by autopart
+        boot_partition = get_device_path_for_mount_point("/boot")
+        safe_exec_program("mount", [boot_partition, self._sysroot + "/boot"])
+        # Make sure the partition is empty
+        for path in glob.glob(self._sysroot + "/boot/*"):
```

Comment:
These 5 lines can be written in one line with:

safe_exec_program("rm", ["-rf", self._sysroot + "/boot/*"])

Reply:
And it was like this before :)